front/lib/api/files/upload.ts

```diff
@@ -509,5 +518,77 @@ export async function uploadToCloudStorage(
     });
   }
 
+  // Upsert to the conversation datasource & generate a snippet.
+  // TODO(JIT) the tool output flow do not go through this path.
+  const jitEnabled = await isJITActionsEnabled(auth);
```

Comment:
Ideally we move that to its own function and call it from the caller of this one since this function is meant to upload to cloud storage?

Reply:
We want uploads of files for useCase X/Y/Z to be turned into datasources, I rather have this function make sure it happens from everywhere it's called (already 2 places, soon more as I'll update the actions).

But I agree "uploadToCloudStorage" is not a great name, will update.